extensions-core/datasketches/src/main/java/org/apache/druid/query/aggregation/datasketches/hll/HllSketchBuildAggregatorFactory.java

```diff
@@ -114,4 +118,19 @@ public int getMaxIntermediateSize()
     return HllSketch.getMaxUpdatableSerializationBytes(getLgK(), TgtHllType.valueOf(getTgtHllType()));
   }
 
+  private void validateInputs(@Nullable ColumnCapabilities capabilities)
+  {
+    if (capabilities != null) {
```

Comment:
nit: can use `Types.is(capabilities, ValueType.COMPLEX)` to get rid of outer if statement since it null checks